src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/afcalc/GeneralPloidyExactAFCalculator.java

```diff
@@ -389,79 +390,130 @@ public GenotypesContext subsetAlleles(final VariantContext vc,
         Utils.nonNull(vc, "vc is null");
         Utils.nonNull(allelesToUse, "allelesToUse is null");
 
-        // the genotypes with PLs
-        final GenotypesContext oldGTs = vc.getGenotypes();
+        final GenotypesContext result = GenotypesContext.create();
 
-        // samples
-        final List<String> sampleIndices = oldGTs.getSampleNamesOrderedByName();
+        // Subset genotypes for each sample
+        for (final Genotype g : vc.getGenotypes()) // If it really needs to process order by sample name do so.
+            result.add(subsetGenotypeAlleles(g, allelesToUse, vc, defaultPloidy, assignGenotypes));
 
-        // the new genotypes to create
-        final GenotypesContext newGTs = GenotypesContext.create();
+        return GATKVariantContextUtils.subsetGenotypesContextAD(result, vc, allelesToUse);
+    }
+
+    /**
+     * From a given genotype, extract a given subset of alleles and update genotype PLs and SACs.
+     * @param g                                 genotype to subset
+     * @param allelesToUse                      alleles to subset
+     * @param vc                                variant context with alleles and genotypes
+     * @param defaultPloidy                     ploidy to assume in case that {@code vc} does not contain that information for a sample.
+     * @param assignGenotypes                   true: assign hard genotypes, false: leave as no-call
+     * @return                                  Genotypes with new PLs and SACs
+     */
+    private Genotype subsetGenotypeAlleles(final Genotype g, final List<Allele> allelesToUse, final VariantContext vc, final int defaultPloidy,
```

Comment:
add unit tests to `GeneralPloidyAFCalculationModelUnitTest` to cover this method